src/main/java/michael/Parser.java

```diff
@@ -0,0 +1,133 @@
+            if (input.length() < 6) { // no number given to mark
+                throw new MichaelException("Enter integer position of task on list to mark. "
+                        + "Use command list to check the position of the required task.");
+            }
+
+            int index = Integer.valueOf(input.substring(5));
+            Task target = tasks.get(index - 1);
+            target.doTask();
+            return "Nice! I've marked this task as done:\n" + "  " + target;
+        } else if (input.equals("list")) { // list user inputs thus far
+            String list = "Here are the tasks in your list:\n";
+
+            for (int i = 0; i < tasks.size(); i++) {
+                String elem = String.valueOf(i + 1) + ". " + tasks.get(i) + "\n";
+                list = list.concat(elem);
+            }
+
+            return list.substring(0, list.length() - 1); // substring to remove last line break
+        } else if (input.length() >= 6 && input.substring(0, 6).equals("unmark")) { // unmark a task
+            if (input.length() < 8) { // no number given to unmark
+                throw new MichaelException("Enter integer position of task on list to unmark. "
+                        + "Use command list to check the position of the required task.");
+            }
+
+            int index = Integer.valueOf(input.substring(7));
+            Task target = tasks.get(index - 1);
+            target.undoTask();
+            return "OK, I've marked this task as not done yet:\n" + "  " + target;
+        } else if (input.length() >= 4 && input.substring(0, 4).equals("todo")) { // task of type todo to be added
+            if (input.length() < 6) { // no task given
+                throw new MichaelException("Enter a task to be done.");
+            }
+
+            ToDo curr = new ToDo(input.substring(5));
+            tasks.add(curr);
+            String message = "Got it. I've added this task:\n" + "  " + curr.toString() + "\n"
+                    + "Now you have " + String.valueOf(tasks.size()) + " tasks in the list.";
+            return message;
+        } else if (input.length() >= 8 && input.substring(0, 8).equals("deadline")) { // task of type deadline to be added
+            if (input.length() < 10) { // no deadline task given
+                throw new MichaelException("Enter a valid task with a deadline.");
+            }
+
+            String task = input.substring(9);
+            String[] parts = task.split("/");
+            for (int i = 0; i < parts.length - 1; i++) {
+                String curr = parts[i];
+                parts[i] = curr.substring(0, curr.length() - 1);
+            }
+
+            String deadline = parts[1].substring(3);
+            if (deadline.split("-").length < 3) { // invalid format of date
+                throw new MichaelException("Deadline should be in format YYYY-MM-DD");
+            }
+
+            Deadline curr = new Deadline(parts[0], deadline);
```

Comment:
Can consider abstracting away the commands into a separate command class so as to leave the Parser class with purely handling inputs from users.